app/controllers/api/v0/users/rounds_controller.rb

```diff
@@ -20,12 +20,19 @@ def current_daily_round
 
   def daily_vote
     round = Game.current_daily_round
-    vote = Vote.create!(lat: params[:lat],
-      lon: params[:lon],
+    original_lat = params[:lat]
+    original_lon = params[:lon]
+    vote = Vote.create!(lat: original_lat,
+      lon: original_lon,
       user_id: params[:user_id],
       round_id: round.id)
     vote.process
-    render json: VoteSerializer.new(vote), status: 201
+    if (vote.lat.to_f - original_lat.to_f).abs >=0.5 || (vote.lon.to_f - original_lon.to_f).abs >=2.0
+        vote.destroy
+        render json: { error: 'Your vote is over water. Please pick a new location.' }, status: 400
```

Comment:
I don't love 400 here. It sounds like this is a type of validation error. The request was fine, as in, the server could interpret it, but could not do what the user wanted for whatever reason. 400 means bad request, as in, the server could make sense of it at all. I would be inclined to call this a 422.

https://www.bennadel.com/blog/2434-http-status-codes-for-invalid-data-400-vs-422.htm